types/cordova-plugin-background-download/index.d.ts

```diff
@@ -0,0 +1,92 @@
+// Type definitions for cordova-plugin-background-download 0.5
```

Comment:
So I've noticed the [underlying library](https://github.com/rohitup/background-download/blob/master/www/BackgroundDownloader.js#L42) uses the CommonJS export style but this file is putting all of the types in the global scope. Should these definitions be using the CommonJS style too or do you have some code to convert over to global scope?

Reply:
Thank you for your review!
This is the first .d.ts file I created. I was trying to do my very best and I hoped this would help other people.
You are right: All the types are put in the global scope by my definition.
BackgroundDownloadPromise and Deferral really are only accessable through the return value of DownloadOperation#startAsync. And BackgroundDownloader really is only acessable through BackgroundTransfer.BackgroundDownloader but needs to be newable.
Should I change some of them to interfaces instead of classes?
I would be very grateful if you have an advise how to improve my definition. Or a link to a tutorial to do so.
Thank you very much!

Reply:
After my last change only BackgroundTransfer.BackgroundDownloader and DownloadOperation are accessible, the rest now are interfaces.

Reply:
The definitions are good, I just wasn't familiar with the cordova-plugin-background-download library usage. You can use either classes or interfaces - the recommendation is that if you have constructors then you should use classes.